### PR TITLE
movement improvements

### PR DESCRIPTION
**Pull request by aeryz**

### CODE REVIEW

File: move/move-ibc/sources/ibc.move [hunk 1]
```diff
@@ -340,6 +334,10 @@ module IBC::Core {
         client_state: vector<u8>,
         consensus_state: vector<u8>,
     ) acquires IBCStore, SignerRef {
+        // NOTE(aeryz): At this point, we don't need to have a routing mechanism because it will introduce
+        // additional gas cost. We should only enforce the use of `cometbls` for the `client_type`
+        assert!(string::bytes(&client_type) == &b"cometbls", E_UNKNOWN_CLIENT_TYPE);
```
Comment (benluelo) [inline move/move-ibc/sources/ibc.move:339]: we should make an issue to track this probably, and benchmark how much gas this actually saves

File: move/move-ibc/sources/ibc.move [hunk 2]
```diff
@@ -786,16 +775,15 @@ module IBC::Core {
...
             proof_height,
             proof_init,
             *channel::counterparty_port_id(&counterparty),
             *channel::counterparty_channel_id(&counterparty),
             bcs::to_bytes(&expected_channel)
-        )) {
-            abort(E_INVALID_PROOF)
-        };
+        );
+        assert!(err == 0, E_INVALID_PROOF);
```
Comment (benluelo) [inline move/move-ibc/sources/ibc.move:786]: why not abort with `err` here?

File: move/move-ibc/sources/ics23.move [hunk 3]
```diff
@@ -229,6 +230,31 @@ module IBC::ics23 {
         buf
     }
 
+    public fun decode_membership_proof(buf: vector<u8>): MembershipProof {
```
Comment (benluelo) [inline move/move-ibc/sources/ics23.move:233]: _bcs?

File: move/move-ibc/sources/ibc.move (hunk 2 above)
Comment (aeryz) [reply move/move-ibc/sources/ibc.move:786]: errors are kinda garbage right now, proper error handling will be my next thing after voyager integration

File: move/move-ibc/sources/ics23.move (hunk 3 above)
Comment (aeryz) [reply move/move-ibc/sources/ics23.move:233]: yeah bcs will probably we a better naming, will do a naming refactor too

File: move/move-ibc/sources/ibc.move (hunk 1 above)
Comment (aeryz) [reply move/move-ibc/sources/ibc.move:339]: #2919

**Review by benluelo — APPROVED** [review]

**Pull request merged**